send.php

```diff
@@ -67,19 +67,32 @@
 $total = 0;
 $successful = 0;
 foreach ($lines as $line) {
-  if (!trim($line)) continue;
-  $payload = json_decode($line, true);
-  $dt = new DateTime($payload["timestamp"]);
-  $ts = floatval($dt->getTimestamp() . "." . $dt->format("u"));
-  $payload["timestamp"] = $ts;
-  $type = $payload["type"];
-  $ret = call_user_func_array(array("Segment", $type), array($payload));
-  if ($ret) $successful++;
-  $total++;
-  if ($total % 100 === 0) Segment::flush();
+    if (!trim($line)) continue;
+    $total++;
+    $payload = json_decode($line, true);
+    $dt = new DateTime($payload["timestamp"]);
+    $ts = floatval($dt->getTimestamp() . "." . $dt->format("u"));
+    $payload["timestamp"] = date("c", (int) $ts);
+    $type = $payload["type"];
+    $currentBatch[] = $payload;
+    // flush before batch gets too big
+    if (mb_strlen((json_encode(array('batch' => $currentBatch, 'sentAt' => date("c")))), '8bit') >= 512000) {
+        $libCurlResponse = Segment::flush();
+        if ($libCurlResponse) {
+            $successful += count($currentBatch) - 1;
+        } else {
+            // todo: maybe write batch to analytics-error.log for more controlled errorhandling
+        }
+        $currentBatch = array();
+    }
+    $payload["timestamp"] = $ts;
```

Comment:
Only question I have is why is this being re-assigned here?

Reply:
Because Client.php automatically transforms the messages timestamp to $payload["timestamp"] = date("c", (int) $ts), but if you directly enqueue the date on that format it converts it to something else(weird behaviour of "private function formatTime"...)
So in order to be able to check the exact size on send.php & Client.php I manually set it to date("c", (int) $ts) and after checking batch size its reassigned to its original value of $ts so the Client.php casts it to date("c",$ts) again (as I said on top if I pass the timestamp formatted with date("c") the formatTime method modifies its value)...

It's kinda ugly but I didn't want to break any library behavior so I "tricked" the send.php file as a pragmatic approach